vector/src/main/scala/geotrellis/vector/io/shape/reader/ShapeDBaseFileReader.scala

```diff
@@ -0,0 +1,237 @@
+package geotrellis.vector.io.shape.reader
+
+import geotrellis.vector.io.FileSystem
+
+import collection.immutable.HashMap
+
+import java.nio.{ByteBuffer, ByteOrder}
+
+import java.util.{Calendar, Date}
+
+import spire.syntax.cfor._
+
+case class MalformedShapeDBaseFileException(msg: String) extends RuntimeException(msg)
+
+import Charset._
+
+/**
+  * Reads a DBase III file (.dbf) for shape files.
+  */
+object ShapeDBaseFileReader {
+
+  val FileExtension = ".dbf"
+
+  def apply(path: String): ShapeDBaseFileReader =
+    if (path.endsWith(FileExtension)) {
+      val charset = CodePageFileReader(path.take(path.length - 4) +
+        CodePageFileReader.FileExtension).read
+      apply(FileSystem.slurp(path), charset)
+    }
+    else throw new MalformedShapeDBaseFileException(
+      s"Bad file ending (must be $FileExtension)."
+    )
+
+  def apply(bytes: Array[Byte], charset: Charset): ShapeDBaseFileReader =
```

Comment:
Is this the same deal as GeoTiffReader where we're making a reader just to call .read on it?